velox/dwio/common/compression/Compression.h

```diff
@@ -60,20 +60,40 @@ class Decompressor {
   const std::string streamDebugInfo_;
 };
 
+struct CompressionOptions {
+  CompressionOptions() {}
+  union Format {
+    Format() {
+      zlib.windowBits = -15;
+    }
+    struct {
+      int windowBits;
+    } zlib;
+  } format;
+};
+
+static struct CompressionOptions dwrfOrcCompressionOptions;
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param isRawUncompress specify whether to perform a raw uncompress
+ * @param compressedLength the compressed block length for raw uncompress
+ * @param zlibWindowBits window bits to use for zlib decompression
  */
 std::unique_ptr<dwio::common::SeekableInputStream> createDecompressor(
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool isRawUncompress = false,
+    size_t compressedLength = 0,
+    CompressionOptions options = dwrfOrcCompressionOptions);
```

Comment:
This comment is for Line 111 and 112. Would it make sense to wrap these levels in the options too?